plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/CreateApplicationFromTemplateJob.java

```diff
@@ -0,0 +1,111 @@
+import org.jboss.tools.openshift.internal.common.core.job.AbstractDelegatingMonitorJob;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+import org.jboss.tools.openshift.internal.ui.dialog.ResourceSummaryDialog;
+import org.jboss.tools.openshift.internal.ui.util.MessageDialogUtil;
+import org.jboss.tools.openshift.internal.ui.wizard.application.IResourceLabelsPageModel.Label;
+
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.capability.CapabilityVisitor;
+import com.openshift.restclient.capability.resources.IProjectTemplateProcessing;
+import com.openshift.restclient.model.IConfig;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.template.IParameter;
+import com.openshift.restclient.model.template.ITemplate;
+
+/**
+ * A Job to create an application from a template
+ */
+public class CreateApplicationFromTemplateJob extends AbstractDelegatingMonitorJob {
+
+	private IProject project;
+	private ITemplate template;
+	private Collection<Label> labels;
+	private Collection<IParameter> parameters;
+
+	public CreateApplicationFromTemplateJob(IProject project, ITemplate template, Collection<IParameter> parameters, Collection<Label> labels) {
+		super("Create Application From Template");
+		this.project = project;
+		this.template = template;
+		this.labels = labels;
+		this.parameters = parameters;
+	}
+
+	@Override
+	protected IStatus doRun(IProgressMonitor monitor) {
+		template.updateParameterValues(parameters);
+		for (Label label : labels) {
+			template.addLabel(label.getName(), label.getValue());
+		}
+		
+		IStatus status = project.accept(new CapabilityVisitor<IProjectTemplateProcessing, IStatus>() {
+
+			@Override
+			public IStatus visit(IProjectTemplateProcessing capability) {
+				
+				try {
+					IConfig config = capability.process(template);
+					
+					Collection<IResource> resources = capability.apply(config);
+					return handleResponse(resources);
+				}catch(OpenShiftException e) {
+					String message = e.getMessage();
+					if(e.getStatus() != null) {
+						message = e.getStatus().getMessage();
+					}
+					showErrorDialog(message);
+					return new Status(IStatus.ERROR, OpenShiftUIActivator.PLUGIN_ID, -1, message, e);
+				}
+			}
+		},
+		new Status(IStatus.INFO, OpenShiftUIActivator.PLUGIN_ID, "Template Processing Capability is Unsupported", null));
+		
+		
+		return status;
+	}
+	
+	private IStatus handleResponse(Collection<IResource> resources) {
+		int severity = IStatus.OK;
+		for (IResource resource : resources) {
+			if(resource.getKind() == ResourceKind.Status) {
+				severity = IStatus.WARNING;
+				break;
+			}
+		}
+		showResourcesDialog(resources);
+		return new Status(severity, OpenShiftUIActivator.PLUGIN_ID, OK,"",null);
+	}
+	
+	public void showResourcesDialog(final Collection<IResource> resources) {
```

Comment:
Ideally you would not want to open a dialog from within a job. A job is run in the Eclipse thread-pool and opening a dialog would gather the (single) UI thread lock and thus block the execution and disposal of the job until the user closed the dialog (and thus possibly blocking further jobs from running). A better pattern is to show the jobs-result in a dialog after the job finished.